source/tests/pt/test_weighted_avg.py

```
@@ -0,0 +1,180 @@
+                "resnet_dt": True,
+                "seed": 1,
+            },
+            "data_stat_nbatch": 20,
+        }
+        self.config["model"] = deepcopy(model_se_e2_a)
+        self.input_json = "test_dp_test.json"
+        with open(self.input_json, "w") as fp:
+            json.dump(self.config, fp, indent=4)
+        trainer = get_trainer(deepcopy(self.config))
+        model = torch.jit.script(trainer.model)
+        self.tmp_model = tempfile.NamedTemporaryFile(delete=False, suffix=".pth")
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)